src/api/app/models/token/workflow.rb

```diff
@@ -15,14 +15,34 @@ def call(options)
 
     workflows.each do |workflow|
       workflow.steps.each do |step|
-        run_step_and_report(step, scm_extractor_payload, scm_token)
+        run_step_and_report(step, scm_extractor_payload, scm_token, workflow.filters)
       end
     end
   end
 
   private
 
-  def run_step_and_report(step, scm_extractor_payload, scm_token)
+  def workflow_repositories(target_project_name, filters)
```

Comment:
Isn't this more like a method of `BranchPackageStep`?

Reply:
Even before the changes from this PR, `run_step_and_report` was already doing too much. I agree with you that the code should be moved to the step.